linkerd/app/core/src/serve.rs

```diff
@@ -79,3 +85,7 @@ where
         _ = shutdown => { Ok(()) }
     }
 }
+
+fn is_io(e: &(dyn std::error::Error + 'static)) -> bool {
```

Comment:
nit/TIOLI: i wonder if we'll want this anywhere else (e.g., how are we logging connection errors on the client side currently?). this could maybe go in `linkerd_error`? we should probably hold off on that until we actually need to use this elsewhere, though...